lib/ar_query_matchers/queries/load_counter.rb

```diff
@@ -15,24 +15,15 @@ def self.instrument(&block)
       end
 
       class LoadQueryFilter < Queries::QueryFilter
-        # Matches named SQL operations like the following:
-        # 'User Load'
-        MODEL_LOAD_PATTERN = /\A(?<model_name>[\w:]+) (Load|Exists)\Z/.freeze
-
         # Matches unnamed SQL operations like the following:
-        # "SELECT COUNT(*) FROM `users` ..."
-        MODEL_SQL_PATTERN = /SELECT .* FROM [`"](?<table_name>[^`"]+)[`"]/.freeze
-
-        def filter_map(name, sql)
-          # First check for a `SELECT * FROM` query that ActiveRecord has
-          # helpfully named for us in the payload
-          match = name.match(MODEL_LOAD_PATTERN)
-          return ModelName.new(match[:model_name]) if match
+        # "SELECT * FROM `users` ..."
+        MODEL_SQL_PATTERN = /SELECT (?:(?!SELECT).)* FROM [`"](?<table_name>[^`"]+)[`"]/.freeze
```

Comment:
**Note**: @dlee148 and I think this would only capture correlated subqueries once. But in reality the sub-query is executed multiple times. Counting that as a non-goal.

Reply:
👆 This comment is outdated!